cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeUtils.java

```diff
@@ -150,6 +158,23 @@ public void cancel() {
     };
   }
 
+  /**
+   * Sets the {@link TaskSupport} for the given Scala {@link ParArray} to {@link ThreadPoolTaskSupport}
+   * with a {@link ThreadPoolExecutor} with zero keep alive time. This method is mainly used
+   * by {@link SparkRunnerClassLoader} to set the {@link TaskSupport} for the parallel array used inside the
+   * {@link DStreamGraph} class in spark.
+   */
+  @SuppressWarnings("unused")
+  public static <T> ParArray<T> setTaskSupport(ParArray<T> parArray) {
+    ThreadPoolExecutor executor = new ThreadPoolExecutor(0, Integer.MAX_VALUE, 1,
```

Comment:
If it is 0 seconds, then we cannot have `allowCoreThreadTimeOut` set to `true`, which will leave a thread hanging around after program terminates.